netty/src/test/java/io/grpc/netty/NettyServerHandlerTest.java

```diff
@@ -1346,7 +1368,7 @@ private ByteBuf emptyGrpcFrame(int streamId, boolean endStream) throws Exception
     try {
       return dataFrame(streamId, endStream, buf);
     } finally {
-      buf.release();
+      buf.clear();
```

Comment:
Revert this change.

Reply:
Fixed

Reply:
As per documentation clear() does not release memory: https://github.com/netty/netty/blob/64c52255c073d4a308d3dbb56a12d3cca7692c47/buffer/src/main/java/io/netty/buffer/ByteBuf.java#L467

I tried with your PR and not calling either byteBuf.clear() or byteBuf.release() here and I did not observe the leak report for any of the 3 tests calling _emptyDataFrame()_ viz headersWithErrAndEndStreamReturnErrorButNotThrowNpe, streamErrorShouldNotCloseChannel and clientHalfCloseShouldForwardToStreamListener.

Reply:
I think in the above comment and snippet we are discussing about 3 tests failures with refCnt decrement error if we use byteBuf.release()  instead of byteBuf.clear(), not about memory leakage issue

Reply:
No, my question is why byteBuf.release() is even required when it doesn't show any report leak.

Reply:
you mean to say we need to remove this catch block which is doing the cleanup part, as this was an existing code?

Reply:
No, leave it as is. For some reason I see 
- buf.release() and 
+ buf.clear()
making it appear like it was your change.

Reply:
I agree the `release()` looks like it should go away (the ownership is being transferred to `dataFrame()`, but `clear()` is useless here. Just delete the line?

Reply:
Fixed

Reply:
Fixed